### DISCUSSION TITLE
How to get superscript number citations

### DISCUSSION THREAD

**Discussion by QiangF** [kjambunathan/org-mode-ox-odt#167] [Q&A] [answered]
Some publication require the in paragraph citation number been formatted as superscript. There are other case like this question https://emacs.stackexchange.com/questions/51984/superscript-number-citations-in-org

**Comment by QiangF** [comment]
I find the following works:

    text^{[cite:@key]}

**Comment by kjambunathan** [comment, marked as answer]
> Some publication require the in paragraph citation number been formatted as superscript. There are other case like this question https://emacs.stackexchange.com/questions/51984/superscript-number-citations-in-org

The 'style'--footnote or numeric etc---are determined by the CSL style.  That is,   '*.csl' file will  say I want references and footnotes in such and such manner, and 'citeproc' and 'org' will read that field and typeset things accordingly.  The attached screenshot will clarify what I am saying.

![Screenshot from 2022-03-11 09-29-13](https://user-images.githubusercontent.com/19869864/157799904-79ab933b-19c6-4332-b227-84a4a68fcc56.png)


So, my guess .... is you are using the wrong 'csl' style, and instead pick the right one.

------


That said, citation is now handled by org core libraries, and citeproc library.  So, any issue or change request has to be directly taken with `emacs-orgmode' list or the 'citeproc' author.

I have cursorily looked at how the new infrastructure works, and the only area where I can make personal contribution is improving the ODT styles ...

For the sake of record, the `oc-csl' library uses 'org' as intermediate format for ODT

```elisp
(defun org-cite-csl--output-format (info)
  "Return expected Citeproc's output format.
INFO is the export state, as a property list.  The return value is a symbol
corresponding to one of the output formats supported by Citeproc: `html',
`latex', or `org'."
  (let ((backend (plist-get info :back-end)))
    (cond
     ((org-export-derived-backend-p backend 'html) 'html)
     ((org-export-derived-backend-p backend 'latex) 'latex)
     (t 'org))))
```

An immediate improvement would be for it to use `org-odt` as a formatter (instead of `org` above)

```elisp
(defvar citeproc-fmt--formatters-alist
  `((org-odt . ,(citeproc-formatter-create
		 :rt (citeproc-formatter-fun-create citeproc-fmt--org-odt-alist)
		 :bib #'citeproc-fmt--org-odt-bib-formatter
		 ))
    (html . ,(citeproc-formatter-create
	      :rt (citeproc-formatter-fun-create citeproc-fmt--html-alist)
	      :bib #'citeproc-fmt--html-bib-formatter))
    (csl-test . ,(citeproc-formatter-create
		  :rt (citeproc-formatter-fun-create citeproc-fmt--csl-test-alist)
		  :bib #'citeproc-fmt--html-bib-formatter
		  :no-external-links t))
    (raw . ,(citeproc-formatter-create :rt #'identity :bib (lambda (x _) x)))
    (org . ,(citeproc-formatter-create :rt #'citeproc-fmt--org-format-rt))
    (latex . ,(citeproc-formatter-create
	       :rt (citeproc-formatter-fun-create citeproc-fmt--latex-alist)))
    (plain . ,(citeproc-formatter-create :rt #'citeproc-rt-to-plain
					 :no-external-links t)))
  "Alist mapping supported output formats to formatter structs.")

````

The next possible improvement would be for me to introduce `ox-odt` (instead of 'org-odt') and introduce improved set of styles.
